corehq/motech/repeaters/management/commands/migrate_all_repeaters.py

```diff
@@ -0,0 +1,115 @@
+import sys
+import traceback
+
+from django.core.management import call_command
+from django.core.management.base import BaseCommand
+
+from corehq.motech.repeaters.dbaccessors import get_all_repeater_docs
+from corehq.motech.repeaters.models import SQLRepeater
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+class Command(BaseCommand):
+
+    AUTO_MIGRATE_ITEMS_LIMIT = 2000
+
+    @classmethod
+    def count_items_to_be_migrated(cls):
+        couch_count = len(get_all_repeater_docs())
+        sql_count = SQLRepeater.objects.count()
+        return couch_count - sql_count
+
+    @classmethod
+    def commit_adding_migration(cls):
+        """
+        This should be the merge commit of the pull request that adds the command to the commcare-hq repository.
+        If this is provided, the failure message in migrate_from_migration will instruct users to deploy this
+        commit before running the command.
+        """
+        return None
+
+    @classmethod
+    @skip_on_fresh_install
+    def migrate_from_migration(cls, apps, schema_editor):
+        """
+            Should only be called from within a django migration.
+            Calls sys.exit on failure.
+        """
+        to_migrate = cls.count_items_to_be_migrated()
+        print(f"Found {to_migrate} Repeater documents to migrate.")
+
+        migrated = to_migrate == 0
+        if migrated:
+            return
+
+        command_name = cls.__module__.split('.')[-1]
+        if to_migrate < cls.AUTO_MIGRATE_ITEMS_LIMIT:
+            try:
+                call_command(command_name)
+                remaining = cls.count_items_to_be_migrated()
+                if remaining != 0:
+                    migrated = False
+                    print(f"Automatic migration failed, {remaining} items remain to migrate.")
+                else:
+                    migrated = True
+            except Exception:
+                traceback.print_exc()
+        else:
+            print("Found {} items that need to be migrated.".format(to_migrate))
+            print("Too many to migrate automatically.")
+
+        if not migrated:
+            print(f"""
+                A migration must be performed before this environment can be upgraded to the latest version
+                of CommCareHQ. This migration is run using the management command {command_name}.
+            """)
+            if cls.commit_adding_migration():
+                print(f"""
+                Run the following commands to run the migration and get up to date:
+
+                    commcare-cloud <env> fab setup_limited_release --set code_branch={cls.commit_adding_migration()}
+
+                    commcare-cloud <env> django-manage --release <release created by previous command> {command_name}
+
+                    commcare-cloud <env> deploy commcare
+                """)
+            sys.exit(1)
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--verify-only',
```

Comment:
@millerdev any suggestion on how can we verify that `--verify-only` flag works fine? I have tested it locally and it was working fine but I could not figure out a way that feels right to test it in an automated way. The thing that came to my mind was to patch `_verify_doc` and assert if it was called.

Reply:
Would it work to make another test similar to `test_migrate_all_repeaters_command` with `call_command('migrate_all_repeaters', verify_only=True, log_file=path)`? In the test setup ensure that some of the repeaters are unmigrated (that is, they are not in SQL), and at the end of the test assert that the count of unmigrated repeaters is the same as it was before the command was called.

Additionally, you could use `log_path` (`--log-path=...`) to pass in a file path (created with [`tempfile.mkstemp`](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp)) and have the test check for expected verify output in that file.